tests/deployment/headnodes.py

```diff
@@ -40,10 +40,17 @@ def server_init():
     server.validate_state()
 
 
+@pytest.yield_fixture
```

Comment:
pytest's yield fixtures let you abuse generators to do cleanup (rather than having to explicitly register finalizers). The test gets run where the yield statement is. RequestContext is a context manager, which seemed the safest way to make sure we actually were cleaning up after ourselves, but it makes explicit cleanup callbacks... awkward. The yield fixture solves the control flow issue presented there.